packages/story-editor/src/types/layoutProvider.ts

```diff
@@ -44,8 +53,8 @@ export interface LayoutProviderState {
   actions: {
     closeCarousel: () => void;
     openCarousel: () => void;
-    setScrollOffset: () => void;
-    setZoomLevel: () => void;
-    setZoomSetting: () => void;
+    setScrollOffset: Dispatch<SetStateAction<ScrollOffset>>;
```

Comment:
Hmm, that's not a very clean type. I wonder if there's a better way to type it.

Reply:
In terms of readability?
We could have `interface StateSetter<T>` for example. Thoughts?

Reply:
That would work. There's nothing built-in for this at least.